src/compiler/checker.ts

```diff
@@ -19885,5 +19888,15 @@ namespace ts {
                 return true;
             }
         }
+
+        function getAmbientModules(): Symbol[] {
+            const result: Symbol[] = [];
+            for (const sym in globals) {
+                if (hasProperty(globals, sym) && ambientModuleSymbolRegex.test(sym)) {

```

Comment:
You may want to update this branch from master to take advantage of a recent change around `Map<T>` performance. Anything declared `Map<T>` (such as the symbol table) now has a `null` prototype, so it is no longer necessary to check `hasProperty`.